### ISSUE TITLE
[Lightgbm] How to save models while training

### ISSUE THREAD

**Issue by yangtian62** [Azure/mmlspark#531]
In python api I can write a callback function to save models while trianing. Is there a way to achieve this goal for spark version? And I noticed that there's no eval and trianing loss printed out. Is there any params to control it? Thanks!

**Comment by imatiach-msft** [comment]
"And I noticed that there's no eval and trianing loss printed out" we only added early stopping with evaluation data in latest master recently, it's not released in v0.16.  You can use a custom build however.  I thought I did see training loss printed on the workers by default though.

**Comment by imatiach-msft** [comment]
"callback function to save models while trianing" unfortunately we don't have that yet.  Will look into adding this functionality.  Could you describe the use case?  What will you be using the saved models at each iteration for?  I was thinking of adding that for dynamic allocation (eg if worker dies or is added, use last saved model to restart training).

**Comment by yangtian62** [comment]
@imatiach-msft Thanks for your reply! I just want to save models for let's say every 100 iterations to keep a snapshot. Sometimes I may need to train a model for several days, and it is much safer if intermediate models could be saved, in case some accidents cut the training process off and I end up with nothing. And sometimes the final model is too large, let's say, the best one gets 0.9 accuracy and weights 1G. There's a big chance that some intermediate model could get 0.88 accuracy and weight only 500M. So in some cases the intermediate model could be a better choice.
